data/pools/lsigEvalCache.go

```diff
@@ -0,0 +1,93 @@
+// Copyright (C) 2019 Algorand, Inc.
+// This file is part of go-algorand
+//
+// go-algorand is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// go-algorand is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+package pools
+
+import (
+	"github.com/algorand/go-algorand/data/transactions"
+	"github.com/algorand/go-algorand/protocol"
+)
+
+type lsigEvalSubcache struct {
+	they map[protocol.ConsensusVersion]map[transactions.Txid]error
+}
+
+func (les *lsigEvalSubcache) get(cvers protocol.ConsensusVersion, txid transactions.Txid) (err error, found bool) {
```

Comment:
A common idiom is to have the error be the last return value when returning multiple results.

Reply:
Ah, sorry, I didn't fully understand what's going on here.  I wonder if it warrants a comment saying this err is the cache lookup result..